nexus/reconfigurator/planning/src/planner.rs

```diff
@@ -1831,7 +1831,9 @@ mod test {
         // NOTE: Expunging a disk doesn't immediately delete datasets; see the
         // "decommissioned_disk_cleaner" background task for more context.
         assert_eq!(diff.datasets.removed.len(), 0);
-        assert_eq!(diff.datasets.modified.len(), 0);
+
+        // The disposition has changed from `InService` to `Expunged`
```

Comment:
This is actually a good change! The dataset was actually modified, we just couldn't see it with the reduction in form to the lowest common denominator shared with inventory that doesn't contain disposition.

Reply:
Is it worth adding a check on the contents of `datasets.modified.first()` (mostly for clarity)? E.g., the checks a few lines below this that confirm the modified zone is crucible and it was expunged.

Reply:
Great idea! Added in [14bc84c](https://github.com/oxidecomputer/omicron/pull/7252/commits/14bc84c28574b646dfb053f9bd0f6bf464830b80)